haystack/pipelines/base.py

```diff
@@ -1690,6 +1690,7 @@ def _build_eval_dataframe(
                     df_docs.map_rows = partial(df_docs.apply, axis=1)
                     df_docs.rename(columns={"id": "document_id", "content": "context"}, inplace=True)
                     df_docs["gold_document_ids"] = [gold_document_ids] * len(df_docs)
+                    df_docs["gold_answers"] = [gold_answers] * len(df_docs)
```

Comment:
added for easier analysis, e.g. if `document_relevance_criterion="answer"`

Reply:
Just to clarify this won't be a problem if there are no `gold_answers` in the eval set? For example in the case we use `context` for the relevance criteria gold_answers may not be available.

Reply:
> Thanks @tstadel this looks good to me! I don't fully understand how the code is structured, but I appreciate that the tests for the metrics look correct now.

I agree, the current code can be better structured. As there are no other using parts besides calculate metrics, refactoring it, wouldn't bring too much value. Let's refactor in another PR, probably connecting it to a feature to support custom metrics.